subcommands/keys/tuf_updates_rotate_online_key.go

```diff
@@ -0,0 +1,87 @@
+package keys
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/foundriesio/fioctl/subcommands"
+)
+
+func init() {
+	rotate := &cobra.Command{
+		Use:   "rotate-online-key --role targets|snapshot|timestamp [--txid=<txid>]",
+		Short: "Stage rotation of the online TUF signing key for the Factory",
+		Long: `Stage rotation of the online TUF signing key for the Factory.
+
+The new online signing key will be used in both CI and production TUF root.
+
+When you rotate the TUF online signing key:
+- if there are CI or production targets in your factory, they are re-signed using the new key.
+- if there is an active wave in your factory, the TUF online key rotation is not allowed.
+- the new wave cannot be created until you apply the online keys rotation.
+
+When you apply the online key rotation, these features are temporarily disabled until it succeeds:
+- new CI targets upload (including the targets upload during CI builds).
+- automatic re-signing of expired TUF roles using online keys (both CI and production targets).
+
+Please, plan the TUF online key rotation timing accurately in order to prevent service disruption.`,
+		Example: `
+- Rotate online TUF targets key and re-sign the new TUF root:
+  fioctl keys tuf updates rotate-online-key \
+    --txid=abc --role=targets --keys=tuf-root-keys.tgz --sign
```

Comment:
Why the `--sign` option is needed? I thought that root meta signing is part of the rotation process.
Or this is for the case if a user would like to rotate a few key types (a.g. targets and snapshot) and then sign once.

Reply:
And there is `sign` sub-command.

Reply:
And the `apply` command doesn't do automatic signing.

Reply:
I would rather do signing for each operation or automatically sign on `apply`. I think a user will always forget to run the `sign` command before running `apply`.

Reply:
I think Voldodya's going to say: "We went through this in the design review and all agreed on this flow". I think that's fair. Also - this is *expert* mode. I kind of liked the granularity when I tested this out. I could do each step and also do the `review` subcommand to observe.

Reply:
Thanks @doanac ! You answered 80% of the question.

The other 20% is that for offline targets key rotation, the root.json is normally not signed by the same person who rotates it. Because, from the security standpoint Root Officer versus Targets Officer should be two different persons.

Although it is technically possible that Root Officer rotates the key and hands it over to the Targets Officer - that's a backward workflow. Furthermore, in one of the future releases we'll introduce the HSM support, which will make this backwards behavior impossible.

So, for the `rotate-offline-key`, the signature is optional; but I added the `--sign` argument for the user's convenience, as 80% of users will do it this way. Once I do that, other commands like e.g. `rotate-online-key` have to follow the same pattern to share the common _look and feel_; and we need a standalone `sign` command.